wadm/lib/wadm/api/api_server.ex

```diff
@@ -0,0 +1,174 @@
+defmodule Wadm.Api.ApiServer do
+  @moduledoc """
+  Exposes the API for interacting with wadm over the NATS subject wadm.api.>
+  The following is a list of the operations available in this API (minus the prefix):
+
+  * `apps.get` - Retrieves a summary of the application models stored by this wadm cluster. This will include status information
+  * `app.put` - Adds a new application model
+  * `app.del` - Deletes an application model
+  * `app.deploy` - Deploys an application to a given lattice
+  * `app.undeploy` - Stops managing an application in the given lattice. Non-destructive.
+  * `app.details` - Retrieves a full set of details about an application, including the model
+  * `app.history` - Retrieves an audit history for a particular application
+  """
+  require Logger
+  use Gnat.Server
+
+  alias Wadm.Model.{Validator, AppSpec}
+
+  def request(%{topic: topic, body: body}) do
+    topic
+    |> String.split(".")
+    # wadm
+    |> List.delete_at(0)
+    # api
+    |> List.delete_at(0)
+    |> List.to_tuple()
+    |> handle_request(body)
+  end
+
+  # TODO just mocking this data up for now.
+  # will make it real once we like the shapes
+  defp handle_request({"apps", "get"}, _body) do
+    response =
+      [
+        %{
+          name: "App name",
+          version: "0.1.0",
+          description: "My app",
+          deployment_status: [
+            %{
+              lattice: "default",
+              # :fully_deployed, :fail
+              status: :partially_deployed
+            }
+          ]
+        }
+      ]
+      |> Jason.encode!()
+
+    {:reply, response}
+  end
+
+  defp handle_request({"app", "put"}, body) do
+    # input here is a JSON version of the OAM model
```

Comment:
Just to double check, is OAM what we want to use? Is that going to be too much overhead for what we need wadm to do?